src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,128 @@
-/**
- * Contains metadata relevant to a representation.
- */
-import { Quad } from 'rdf-js';
+import { Store } from 'n3';
+import { BlankNode, Literal, NamedNode, Quad, Term } from 'rdf-js';
+import { quad as createQuad, literal, namedNode } from '@rdfjs/data-model';
 
 /**
- * Metadata corresponding to a {@link Representation}.
+ * Stores the metadata triples and provides methods for easy access.
  */
-export interface RepresentationMetadata {
+export class RepresentationMetadata {
+  private store: Store;
+  private id: NamedNode | BlankNode;
+
   /**
-   * All metadata triples of the resource.
+   * @param identifier - Identifier of the resource relevant to this metadata.
+   *                     A blank node will be generated if none is provided.
+   *                     Strings will be converted to named nodes.
+   * @param quads - Quads to fill the metadata with.
    */
-  raw: Quad[];
+  public constructor(identifier?: NamedNode | BlankNode | string, quads?: Quad[]) {
+    this.store = new Store(quads);
+    if (identifier) {
+      if (typeof identifier === 'string') {
+        this.id = namedNode(identifier);
+      } else {
+        this.id = identifier;
+      }
+    } else {
+      this.id = this.store.createBlankNode();
+    }
+  }
+
   /**
-   * Optional metadata profiles.
+   * @returns All metadata quads.
    */
-  profiles?: string[];
+  public quads(): Quad[] {
+    return this.store.getQuads(null, null, null, null);
+  }
+
   /**
-   * Optional size of the representation.
+   * Identifier of the resource this metadata is relevant to.
+   * Will update all relevant triples if this value gets changed.
    */
-  byteSize?: number;
+  public get identifier(): NamedNode | BlankNode {
+    return this.id;
+  }
+
+  public set identifier(id: NamedNode | BlankNode) {
+    const quads = this.quads().map((quad): Quad => {
+      if (quad.subject.equals(this.id)) {
+        return createQuad(id, quad.predicate, quad.object, quad.graph);
+      }
+      if (quad.object.equals(this.id)) {
+        return createQuad(quad.subject, quad.predicate, id, quad.graph);
+      }
+      return quad;
+    });
```

Comment:
Took me too long to parse this. Can we have a comment preceding this, explaining that we have to change the subjects and objects of triples now?